spec/controllers/target_controller_spec.rb

```diff
@@ -23,10 +25,59 @@
   end
 
   describe 'GET index' do
-    it 'renders a list of targets' do
-      targets = create_list(:target, 2)
-      get :index
-      expect(assigns(:targets)).to match_array(targets)
+    context 'when the user is logged in' do
+      login_user
+
+      it 'renders a list of targets' do
+        targets = create_list(:target, 2)
+        get :index
+        expect(assigns(:targets)).to match_array(targets)
+      end
+
+      context 'JSON format' do
+        login_user
+        let!(:target_1) { create :target }
+        let!(:target_2) { create :target }
```

Comment:
I'm not finding where you are using `target_2`. Are you sure it's necessary?

Reply:
Im using it here ´it 'renders a list of targets' do
          expect(@parsed_body.length).to eq(2)
        end´

Reply:
Ah okay I see. But then since you are not referencing them individually consider doing 
`let!(:targets) { create_list(:target, 2) }`

Reply:
I was doing that, but changed it so I could do this `expect(@parsed_body[0]['edit_url']).to eq(edit_target_url(target_1))`

Reply:
Perfect. Ignore comment then.